{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "687645ff-c312-445f-b542-9511c03e330c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.insert(1, '/Users/labadmin/Projects/immusign')\n",
    "import shutil\n",
    "import os.path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "daf99de0-453e-4b95-bc64-8a4fa9cefc90",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using `tqdm.autonotebook.tqdm` in notebook mode. Use `tqdm.tqdm` instead to force console mode (e.g. in jupyter console)\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from tqdm.notebook import tqdm\n",
    "np.float = float\n",
    "np.bool = bool\n",
    "np.int = int\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import random\n",
    "import torch\n",
    "import time\n",
    "from torch.nn.utils.rnn import pack_padded_sequence, pad_packed_sequence\n",
    "import utils\n",
    "import shap\n",
    "from sklearn.metrics import classification_report\n",
    "from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "857b8991-bbe0-44f2-8f0c-e2ac53982ce6",
   "metadata": {},
   "source": [
    "# Read Filereport"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e7c2264e-79f1-4124-b70d-b2399316b865",
   "metadata": {},
   "outputs": [],
   "source": [
    "filereport = pd.read_csv(\"intega/filereport_read_run_PRJEB55475_tsv.txt\",delim_whitespace=True)\n",
    "filereport[\"filenames\"] = filereport[\"sra_ftp\"].apply(lambda x: \"_\".join(x.split(\"/\")[-1].split(\".\")[0].split(\"_\")[0:2]))\n",
    "filereport[\"stripped_patient_id\"] = filereport[\"filenames\"].apply(utils.get_stripped_pat_no)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "29cc1220-b329-46ac-b9b2-09c2b80ddfd0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8b9a033d95d04f61b0841d49f65126a8",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/146 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for file in tqdm(filereport.filenames):\n",
    "    file_name = file + \".clones.txt\"\n",
    "    dest_path = \"intega/Intega_final/\"\n",
    "    src_path = \"intega/Intega/\"\n",
    "    if not os.path.exists(dest_path + file_name):\n",
    "        shutil.copy2( src_path + file_name, dest_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8c9aec58-be71-4326-b55f-f03077cdb35a",
   "metadata": {},
   "source": [
    "# Load Data read from R-Script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b6e913ee-07fc-4cf3-a4f0-f44abe486f23",
   "metadata": {},
   "outputs": [],
   "source": [
    "import rpy2.robjects as robjects\n",
    "robjects.r['load'](\"intega/final_intega_40000_reads_with_out_of_frame.RData\")\n",
    "twb = robjects.r['twb']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a7170f07-0642-4591-bc27-b1eebf443835",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = utils.convert_rtwb_to_pdtwb(twb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "64f6a6ed-3bc2-46e3-b42e-7387dbb01098",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"clones.txt.name\"] = df[\"sample\"].apply(lambda x: x + \".clones.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "54a3b449-3199-401a-8ef0-526331106d25",
   "metadata": {},
   "source": [
    "# Add metric info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "332e03d9-f0fd-4a73-a615-d25870ec7082",
   "metadata": {},
   "outputs": [],
   "source": [
    "stats = pd.read_csv(\"intega/output/stats.csv\", sep=\";\")\n",
    "stats.rename(columns={stats.columns[0]:\"sample\"}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d7226f17-a456-41c0-9f31-10e6920bd270",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.merge(stats.iloc[:-2], on = \"sample\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "8f301077-1815-4798-ae99-f220ce78bd93",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "132"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df[\"clones.txt.name\"].unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c77c458-beca-4cf8-95cd-36baedbbb221",
   "metadata": {},
   "source": [
    "# Get meta info"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b88e061-d1a1-40ed-92db-7f268c12562b",
   "metadata": {},
   "source": [
    "## get disposition data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "786ae9e3-574c-4954-b54e-3ec0f5ac07a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "xl_file = pd.ExcelFile(\"intega/disposition_2022-07-01.xlsx\")\n",
    "\n",
    "dfs = {sheet_name: xl_file.parse(sheet_name) \n",
    "          for sheet_name in xl_file.sheet_names}\n",
    "disposition = dfs[\"disposition\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fd88d01b-7cd7-4a77-87c4-b66cced806bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Pat-No.', 'Age [y]', 'Gender (M/F)', 'Random No.',\n",
       "       'Treatment as Randomized', 'Treatment as Treated',\n",
       "       'Prior Surgery (Yes/No)', 'HER2 Status Local (IHC 2+/3+)',\n",
       "       'HER2 Status Central', 'Tumor Proportion Score', 'Immune Cell Score',\n",
       "       'Combined Positive Score', 'Prior Drug Therapy',\n",
       "       'Therapy Duration, Any Component [mo]',\n",
       "       'Therapy Duration, All Scheduled Components [mo]',\n",
       "       'Therapy Duration, Nivolumab [mo]', 'Reason for End of Treatment',\n",
       "       'Overall Best Response (CR/PR)', 'Time to Best Response [mo]',\n",
       "       'Response at 1st Tumor Assessment (PD/SD/PR/CR/NA)',\n",
       "       'Sum of Target Lesion Diameters [mm] at Screening',\n",
       "       '% Change of Diameters (target lesions) at 1st Tumor Assessment',\n",
       "       'Best % Change of Diameters (Taget Lesions) during Tumor Assessments',\n",
       "       'PFS [mo]', 'Censoring (Yes/No)', 'No of AEs (Grade 3-5)', 'No of SAEs',\n",
       "       'OS [mo]', 'Death (Yes/No)'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disposition.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2a2594b-27ba-4e35-b0c6-d8b0da353137",
   "metadata": {},
   "source": [
    "## get results data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "858f917d-eb45-4b59-81b8-ca2c58425019",
   "metadata": {},
   "outputs": [],
   "source": [
    "xl_file = pd.ExcelFile(\"intega/Ergebnisse_CDC_Her2_Aug2021_TRB.Immun_Metriken_INTEGA.xlsx\")\n",
    "\n",
    "dfs = {sheet_name: xl_file.parse(sheet_name) \n",
    "          for sheet_name in xl_file.sheet_names}\n",
    "Ergebnisse_CDC_Her2 = dfs[\"Ergebnisse_CDC_Her2_Aug2021_INT\"]\n",
    "Ergebnisse_CDC_Her2.columns = Ergebnisse_CDC_Her2.iloc[0]\n",
    "Ergebnisse_CDC_Her2 = Ergebnisse_CDC_Her2.iloc[1:].copy()\n",
    "\n",
    "\n",
    "new_colums = [ 'Zentrum',              'Pat-ID',\n",
    "                'Patient-ID',               'Probe',                   np.nan,\n",
    "             'Probenabnahme',       'Datum_Analyse',                'CTC1',\n",
    "                    'HER2-0',              'HER2-1',              'HER2-2',\n",
    "                    'HER2-3',                 'CXC',             'PD-L1-0',\n",
    "                   'PD-L1-1',             'PD-L1-2',             'PD-L1-3',\n",
    "                'Kommentare1',                   \"Kommentare2\",                   \"Kommentare3\",\n",
    "                         np.nan,                  'ID',           'all.names',\n",
    "                 'Clonality',               'Group',           'Diversity',\n",
    "                     'Group',            'Richness',               'Group']\n",
    "Ergebnisse_CDC_Her2.columns = new_colums"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "dc7d9f37-818d-4bad-b521-1d994997120c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Zentrum</th>\n",
       "      <th>Pat-ID</th>\n",
       "      <th>Patient-ID</th>\n",
       "      <th>Probe</th>\n",
       "      <th>NaN</th>\n",
       "      <th>Probenabnahme</th>\n",
       "      <th>Datum_Analyse</th>\n",
       "      <th>CTC1</th>\n",
       "      <th>HER2-0</th>\n",
       "      <th>HER2-1</th>\n",
       "      <th>...</th>\n",
       "      <th>Kommentare3</th>\n",
       "      <th>NaN</th>\n",
       "      <th>ID</th>\n",
       "      <th>all.names</th>\n",
       "      <th>Clonality</th>\n",
       "      <th>Group</th>\n",
       "      <th>Diversity</th>\n",
       "      <th>Group</th>\n",
       "      <th>Richness</th>\n",
       "      <th>Group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0001-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2018-03-08 00:00:00</td>\n",
       "      <td>2018-03-13 00:00:00</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BL29</td>\n",
       "      <td>0001-001</td>\n",
       "      <td>Svenja-TRB-0001-001-BL-INTEGA_S100</td>\n",
       "      <td>0.094284</td>\n",
       "      <td>BL</td>\n",
       "      <td>9.551617</td>\n",
       "      <td>BL</td>\n",
       "      <td>1495</td>\n",
       "      <td>BL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0001-001</td>\n",
       "      <td>1Tag2Zyk</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2018-04-05 00:00:00</td>\n",
       "      <td>2018-04-09 00:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C24</td>\n",
       "      <td>0001-001</td>\n",
       "      <td>Svenja-TRB-0001-001-C2-INTEGA_S101</td>\n",
       "      <td>0.08677</td>\n",
       "      <td>C2</td>\n",
       "      <td>9.816343</td>\n",
       "      <td>C2</td>\n",
       "      <td>1721</td>\n",
       "      <td>C2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0001-001</td>\n",
       "      <td>1Tag4-5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2018-06-04 00:00:00</td>\n",
       "      <td>2018-06-06 00:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0001-001</td>\n",
       "      <td>ProEnd</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2018-11-29 00:00:00</td>\n",
       "      <td>2018-11-30 00:00:00</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0001-002</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-02-22 00:00:00</td>\n",
       "      <td>2019-02-26 00:00:00</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BL23</td>\n",
       "      <td>0001-002</td>\n",
       "      <td>Svenja-TRB-0001-002-BL-INTEGA_S79</td>\n",
       "      <td>0.082909</td>\n",
       "      <td>BL</td>\n",
       "      <td>9.154001</td>\n",
       "      <td>BL</td>\n",
       "      <td>1011</td>\n",
       "      <td>BL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>38</td>\n",
       "      <td>5</td>\n",
       "      <td>0038-005</td>\n",
       "      <td>1Tag4-5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2020-04-01 00:00:00</td>\n",
       "      <td>2020-04-06 00:00:00</td>\n",
       "      <td>142</td>\n",
       "      <td>130</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0038-005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.124895</td>\n",
       "      <td>C2</td>\n",
       "      <td>8.294035</td>\n",
       "      <td>C2</td>\n",
       "      <td>713</td>\n",
       "      <td>C2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>0039-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-11-21 00:00:00</td>\n",
       "      <td>2019-11-22 00:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0039-001</td>\n",
       "      <td>NC-hs-TRB-0039-001-Baseline-PB-gDNA_S66</td>\n",
       "      <td>0.053723</td>\n",
       "      <td>BL</td>\n",
       "      <td>10.474131</td>\n",
       "      <td>BL</td>\n",
       "      <td>2148</td>\n",
       "      <td>BL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>0039-001</td>\n",
       "      <td>ProEnd</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2020-01-28 00:00:00</td>\n",
       "      <td>2020-01-30 00:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>?</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>03-019</td>\n",
       "      <td>NC-hs-TRB-03-019-Baseline-PB-gDNA_S25</td>\n",
       "      <td>0.63333</td>\n",
       "      <td>BL</td>\n",
       "      <td>2.97887</td>\n",
       "      <td>BL</td>\n",
       "      <td>279</td>\n",
       "      <td>BL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>230 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Zentrum Pat-ID Patient-ID     Probe  NaN        Probenabnahme  \\\n",
       "1         1      1   0001-001  Baseline  1.0  2018-03-08 00:00:00   \n",
       "2         1      1   0001-001  1Tag2Zyk  2.0  2018-04-05 00:00:00   \n",
       "3         1      1   0001-001   1Tag4-5  3.0  2018-06-04 00:00:00   \n",
       "4         1      1   0001-001    ProEnd  4.0  2018-11-29 00:00:00   \n",
       "5         1      2   0001-002  Baseline  1.0  2019-02-22 00:00:00   \n",
       "..      ...    ...        ...       ...  ...                  ...   \n",
       "226      38      5   0038-005   1Tag4-5  3.0  2020-04-01 00:00:00   \n",
       "227      39      1   0039-001  Baseline  1.0  2019-11-21 00:00:00   \n",
       "228     NaN    NaN        NaN       NaN  NaN                  NaN   \n",
       "229      39      1   0039-001    ProEnd  4.0  2020-01-28 00:00:00   \n",
       "230     NaN    NaN          ?       NaN  NaN                  NaN   \n",
       "\n",
       "           Datum_Analyse CTC1 HER2-0 HER2-1  ... Kommentare3   NaN        ID  \\\n",
       "1    2018-03-13 00:00:00    1    NaN    NaN  ...         NaN  BL29  0001-001   \n",
       "2    2018-04-09 00:00:00    0    NaN    NaN  ...         NaN   C24  0001-001   \n",
       "3    2018-06-06 00:00:00    0    NaN    NaN  ...         NaN   NaN       NaN   \n",
       "4    2018-11-30 00:00:00    1      1    NaN  ...         NaN   NaN       NaN   \n",
       "5    2019-02-26 00:00:00    5      5    NaN  ...         NaN  BL23  0001-002   \n",
       "..                   ...  ...    ...    ...  ...         ...   ...       ...   \n",
       "226  2020-04-06 00:00:00  142    130     12  ...         NaN   NaN  0038-005   \n",
       "227  2019-11-22 00:00:00    0    NaN    NaN  ...         NaN   NaN  0039-001   \n",
       "228                  NaN  NaN    NaN    NaN  ...         NaN   NaN       NaN   \n",
       "229  2020-01-30 00:00:00    0    NaN    NaN  ...         NaN   NaN       NaN   \n",
       "230                  NaN  NaN    NaN    NaN  ...         NaN   NaN    03-019   \n",
       "\n",
       "                                   all.names Clonality Group  Diversity Group  \\\n",
       "1         Svenja-TRB-0001-001-BL-INTEGA_S100  0.094284    BL   9.551617    BL   \n",
       "2         Svenja-TRB-0001-001-C2-INTEGA_S101   0.08677    C2   9.816343    C2   \n",
       "3                                        NaN       NaN   NaN        NaN   NaN   \n",
       "4                                        NaN       NaN   NaN        NaN   NaN   \n",
       "5          Svenja-TRB-0001-002-BL-INTEGA_S79  0.082909    BL   9.154001    BL   \n",
       "..                                       ...       ...   ...        ...   ...   \n",
       "226                                      NaN  0.124895    C2   8.294035    C2   \n",
       "227  NC-hs-TRB-0039-001-Baseline-PB-gDNA_S66  0.053723    BL  10.474131    BL   \n",
       "228                                      NaN       NaN   NaN        NaN   NaN   \n",
       "229                                      NaN       NaN   NaN        NaN   NaN   \n",
       "230    NC-hs-TRB-03-019-Baseline-PB-gDNA_S25   0.63333    BL    2.97887    BL   \n",
       "\n",
       "    Richness Group  \n",
       "1       1495    BL  \n",
       "2       1721    C2  \n",
       "3        NaN   NaN  \n",
       "4        NaN   NaN  \n",
       "5       1011    BL  \n",
       "..       ...   ...  \n",
       "226      713    C2  \n",
       "227     2148    BL  \n",
       "228      NaN   NaN  \n",
       "229      NaN   NaN  \n",
       "230      279    BL  \n",
       "\n",
       "[230 rows x 29 columns]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Ergebnisse_CDC_Her2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e22cc4a0-180c-47db-8cd9-b94def75c548",
   "metadata": {},
   "source": [
    "## get probe review"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "id": "8172f6d5-6d04-41fd-9e88-664356f6f32d",
   "metadata": {},
   "outputs": [],
   "source": [
    "xl_file = pd.ExcelFile(\"intega/Probenübersicht Sequenzierungsstatus Intega-Studie alle Zeitpunkte_LP.xlsx\")\n",
    "\n",
    "proben_df = {sheet_name: xl_file.parse(sheet_name) \n",
    "          for sheet_name in xl_file.sheet_names}\n",
    "proben_df = proben_df[\"Probe_vorhanden\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "6c7f9604-86c8-4175-8d8f-19e3c327e021",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Status', 'Patient', 'Tumorproben', 'Tumor', 'BL', 'C2', 'C5', 'EOT/PD',\n",
       "       'Notizen', 'Cycle_EOT', 'Datum BL', 'Datum C2_sample',\n",
       "       'Datum C5_sample', 'Datum EOT_sample', 'Datum_EOT_real',\n",
       "       'week C2 sample', 'week C5 sample', 'week EOT sample',\n",
       "       'week_to_EOT_C2_sample', 'week_to_EOT_C5_sample',\n",
       "       'week_to_EOT_EOT_sample', 'Unnamed: 21', 'ja: Probe vorhanden'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 267,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proben_df.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42b4aeb0-e590-4820-865e-2e50656e917a",
   "metadata": {},
   "source": [
    "# Merge meta info"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "887b09fa-22cc-4210-b856-110f17fb417d",
   "metadata": {},
   "source": [
    "## merge dispistion "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "74ea6fae-8d46-4264-b70b-e96f342e6dc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"stripped_patient_id\"] = df[\"sample\"].apply(utils.get_stripped_pat_no)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "fe08ce05-b56b-4568-a6c9-ece60665a70e",
   "metadata": {},
   "outputs": [],
   "source": [
    "disposition[\"stripped_patient_id\"] = disposition[\"Pat-No.\"].apply(utils.get_stripped_pat_no)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "2534dc50-74ab-4959-ae93-47c44f3eb568",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Patients not in disposition:  {'38-1'}\n",
      "Patients not in twb data:  {'15-4', '22-3', '16-7', '1-3', '32-4', '6-1', '35-5', '2-4'}\n"
     ]
    }
   ],
   "source": [
    "print(\"Patients not in disposition: \", set(df[\"stripped_patient_id\"].unique()).difference(set(disposition[\"stripped_patient_id\"].unique())))\n",
    "print(\"Patients not in twb data: \", set(disposition[\"stripped_patient_id\"].unique()).difference(set(df[\"stripped_patient_id\"].unique())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "c252136a-fd94-4d7d-aec0-44f4e3879d58",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.merge(disposition, on = \"stripped_patient_id\", how = \"left\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dcc731e6-1a8f-4f2d-8538-9db318f1a653",
   "metadata": {},
   "source": [
    "## merge results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "20c4d10e-8c36-4087-8dcf-a78e1744c726",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([      'Zentrum',        'Pat-ID',    'Patient-ID',         'Probe',\n",
       "                   nan, 'Probenabnahme', 'Datum_Analyse',          'CTC1',\n",
       "              'HER2-0',        'HER2-1',        'HER2-2',        'HER2-3',\n",
       "                 'CXC',       'PD-L1-0',       'PD-L1-1',       'PD-L1-2',\n",
       "             'PD-L1-3',   'Kommentare1',   'Kommentare2',   'Kommentare3',\n",
       "                   nan,            'ID',     'all.names',     'Clonality',\n",
       "               'Group',     'Diversity',         'Group',      'Richness',\n",
       "               'Group'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Ergebnisse_CDC_Her2.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "8546ea68-ab05-4acd-b924-c2a8fd3a182b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'None'"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "str(None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "bcabd1d6-6760-4ecf-9670-d2b8dfff0f7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Ergebnisse_CDC_Her2[\"stripped_patient_id\"] = Ergebnisse_CDC_Her2[\"all.names\"].apply(utils.get_stripped_pat_no)\n",
    "Ergebnisse_CDC_Her2[\"Probe_cleaned\"] = Ergebnisse_CDC_Her2[\"Probe\"].apply(lambda x: \"BL\" if x in [\"Baseline\", \"BL\"] \n",
    "                                                                          else ( \"C2\" if \"2Zyk\" in str(x) else x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "e46a4395-5efb-4cc7-a28f-4b6f8bf8fac2",
   "metadata": {},
   "outputs": [],
   "source": [
    "Ergebnisse_CDC_Her2.loc[Ergebnisse_CDC_Her2[\"stripped_patient_id\"].isnull(), \"stripped_patient_id\"] = Ergebnisse_CDC_Her2.loc[Ergebnisse_CDC_Her2[\"stripped_patient_id\"].isnull(), \"Patient-ID\"].apply(utils.get_stripped_pat_no)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "02599cb2-ad59-4685-8e2a-8b9f469e83d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Patients not in results:  set()\n",
      "Patients not in twb data:  {'3-19', '22-3', '9-1', '16-7', '1-3', '32-4', '1-6', '6-1', '15-8', None, '2-4'}\n"
     ]
    }
   ],
   "source": [
    "print(\"Patients not in results: \", set(df[\"stripped_patient_id\"].unique()).difference(set(Ergebnisse_CDC_Her2[\"stripped_patient_id\"].unique())))\n",
    "print(\"Patients not in twb data: \", set(Ergebnisse_CDC_Her2[\"stripped_patient_id\"].unique()).difference(set(df[\"stripped_patient_id\"].unique())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "c569a46b-e5c8-4675-ac2e-b5a0b7017216",
   "metadata": {},
   "outputs": [],
   "source": [
    "Ergebnisse_per_probe_patient = Ergebnisse_CDC_Her2.groupby([\"Probe_cleaned\", \"stripped_patient_id\"]).apply(len).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "1a50161a-eb61-40d2-be3f-4903b1b48e0e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Probe_cleaned</th>\n",
       "      <th>stripped_patient_id</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1Tag4-5</td>\n",
       "      <td>16-3</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1Tag4-5</td>\n",
       "      <td>38-1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>BL</td>\n",
       "      <td>12-1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>BL</td>\n",
       "      <td>2-11</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>BL</td>\n",
       "      <td>27-2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>BL</td>\n",
       "      <td>6-2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>BL</td>\n",
       "      <td>7-6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>C2</td>\n",
       "      <td>16-3</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>C2</td>\n",
       "      <td>2-10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>C2</td>\n",
       "      <td>2-5</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>C2</td>\n",
       "      <td>2-9</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>C2</td>\n",
       "      <td>5-4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>C2</td>\n",
       "      <td>7-3</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>C2</td>\n",
       "      <td>7-4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>C2</td>\n",
       "      <td>7-5</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Probe_cleaned stripped_patient_id  0\n",
       "6         1Tag4-5                16-3  2\n",
       "20        1Tag4-5                38-1  2\n",
       "32             BL                12-1  2\n",
       "54             BL                2-11  2\n",
       "77             BL                27-2  2\n",
       "103            BL                 6-2  2\n",
       "108            BL                 7-6  2\n",
       "126            C2                16-3  2\n",
       "131            C2                2-10  2\n",
       "140            C2                 2-5  2\n",
       "144            C2                 2-9  2\n",
       "169            C2                 5-4  2\n",
       "175            C2                 7-3  2\n",
       "176            C2                 7-4  2\n",
       "177            C2                 7-5  2"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Ergebnisse_per_probe_patient[Ergebnisse_per_probe_patient[0] > 1] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "1d263824-cfb5-42a5-afb8-c086d1bb3fa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "double_probe_patient = Ergebnisse_per_probe_patient[Ergebnisse_per_probe_patient[0] > 1][[\"Probe_cleaned\", \"stripped_patient_id\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c962589b-7aae-48f6-8c18-3cf1b6a0796a",
   "metadata": {},
   "source": [
    "### mehrere ergebnisse für gleiche patienten --> filereport_read_run_PRJEB55475_tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "2ff346ca-f764-4852-a209-c84101c06453",
   "metadata": {},
   "outputs": [],
   "source": [
    "filereport = pd.read_csv(\"intega/filereport_read_run_PRJEB55475_tsv.txt\",delim_whitespace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "8cb0e8ec-1439-4b7a-b5c4-9bcaff832f1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run_accession</th>\n",
       "      <th>sample_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>study_accession</th>\n",
       "      <th>tax_id</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>fastq_ftp</th>\n",
       "      <th>submitted_ftp</th>\n",
       "      <th>sra_ftp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERR10747912</td>\n",
       "      <td>SAMEA112286929</td>\n",
       "      <td>ERX10200346</td>\n",
       "      <td>PRJEB55475</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo</td>\n",
       "      <td>sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/012/ERR107...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747912/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERR10747914</td>\n",
       "      <td>SAMEA112286931</td>\n",
       "      <td>ERX10200348</td>\n",
       "      <td>PRJEB55475</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo</td>\n",
       "      <td>sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/014/ERR107...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747914/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERR10747915</td>\n",
       "      <td>SAMEA112286932</td>\n",
       "      <td>ERX10200349</td>\n",
       "      <td>PRJEB55475</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo</td>\n",
       "      <td>sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/015/ERR107...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747915/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ERR10747916</td>\n",
       "      <td>SAMEA112286933</td>\n",
       "      <td>ERX10200350</td>\n",
       "      <td>PRJEB55475</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo</td>\n",
       "      <td>sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/016/ERR107...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747916/...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ERR10747919</td>\n",
       "      <td>SAMEA112286936</td>\n",
       "      <td>ERX10200353</td>\n",
       "      <td>PRJEB55475</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo</td>\n",
       "      <td>sapiens</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/019/ERR107...</td>\n",
       "      <td>ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747919/...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  run_accession sample_accession experiment_accession study_accession  tax_id  \\\n",
       "0   ERR10747912   SAMEA112286929          ERX10200346      PRJEB55475    9606   \n",
       "1   ERR10747914   SAMEA112286931          ERX10200348      PRJEB55475    9606   \n",
       "2   ERR10747915   SAMEA112286932          ERX10200349      PRJEB55475    9606   \n",
       "3   ERR10747916   SAMEA112286933          ERX10200350      PRJEB55475    9606   \n",
       "4   ERR10747919   SAMEA112286936          ERX10200353      PRJEB55475    9606   \n",
       "\n",
       "  scientific_name fastq_ftp  \\\n",
       "0            Homo   sapiens   \n",
       "1            Homo   sapiens   \n",
       "2            Homo   sapiens   \n",
       "3            Homo   sapiens   \n",
       "4            Homo   sapiens   \n",
       "\n",
       "                                       submitted_ftp  \\\n",
       "0  ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/012/ERR107...   \n",
       "1  ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/014/ERR107...   \n",
       "2  ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/015/ERR107...   \n",
       "3  ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/016/ERR107...   \n",
       "4  ftp.sra.ebi.ac.uk/vol1/fastq/ERR107/019/ERR107...   \n",
       "\n",
       "                                             sra_ftp  \n",
       "0  ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747912/...  \n",
       "1  ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747914/...  \n",
       "2  ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747915/...  \n",
       "3  ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747916/...  \n",
       "4  ftp.sra.ebi.ac.uk/vol1/run/ERR107/ERR10747919/...  "
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filereport.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "bcf9a368-f344-41f0-bb83-f9545b4bb8b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "filereport[\"filenames\"] = filereport[\"sra_ftp\"].apply(lambda x: x.split(\"/\")[-1].split(\".\")[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "fe5c503d-426c-40b6-a28d-5863eb47f0ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "filereport[\"stripped_patient_id\"] = filereport[\"filenames\"].apply(utils.get_stripped_pat_no)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "05605418-2c99-435b-bf95-372eb2101ea2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16-3 1Tag4-5\n",
      "['NC-hs-TRB-0016-003-2-Zyklus-PB-gDNA_S32_L001_R2_001', 'Dona-hs-TRB-016-003-PB-Baseline-gDNA_S124_L001_R2_001']\n",
      "\n",
      "\n",
      "38-1 1Tag4-5\n",
      "['NC-hs-TRB-0038-001-Baseline-PB-gDNA_S11_L001_R2_001', 'NC-hs-TRB-0038-001-2-Zyklus-PB-gDNA_S12_L001_R2_001']\n",
      "\n",
      "\n",
      "12-1 BL\n",
      "['NC-hs-TRB-0012-001-Baseline-PB-gDNA_S23_L001_R2_001', 'Svenja-TRB-0012-001-C2-INTEGA_S119_L001_R2_001']\n",
      "\n",
      "\n",
      "2-11 BL\n",
      "['NC-hs-TRB-0002-011-Baseline-PB-gDNA_S16_L001_R2_001', 'Svenja-TRB-0002-011-C2-INTEGA_S153_L001_R2_001']\n",
      "\n",
      "\n",
      "27-2 BL\n",
      "['Svenja-TRB-0027-002-C2-INTEGA_S141_L001_R2_001', 'Svenja-TRB-0027-002-BL-INTEGA_S140_L001_R2_001']\n",
      "Match:  Svenja-TRB-0027-002-BL-INTEGA_S140_L001_R2_001\n",
      "\n",
      "\n",
      "6-2 BL\n",
      "['NC-hs-TRB-0006-002-Baseline-PB-gDNA_S18_L001_R2_001']\n",
      "\n",
      "\n",
      "7-6 BL\n",
      "['NC-hs-TRB-0007-006-Baseline-PB-gDNA_S22_L001_R2_001']\n",
      "\n",
      "\n",
      "16-3 C2\n",
      "['NC-hs-TRB-0016-003-2-Zyklus-PB-gDNA_S32_L001_R2_001', 'Dona-hs-TRB-016-003-PB-Baseline-gDNA_S124_L001_R2_001']\n",
      "\n",
      "\n",
      "2-10 C2\n",
      "['Svenja-TRB-0002-010-BL-INTEGA_S152_L001_R2_001', 'NC-hs-TRB-0002-010-2-Zyklus-PB-gDNA_S15_L001_R2_001']\n",
      "\n",
      "\n",
      "2-5 C2\n",
      "['NC-hs-TRB-0002-005-2-Zyklus-PB-gDNA_S13_L001_R2_001', 'Dona-hs-TRB-002-005-PB-Baseline-gDNA_S102_L001_R2_001']\n",
      "\n",
      "\n",
      "2-9 C2\n",
      "['NC-hs-TRB-0002-009-2-Zyklus-PB-gDNA_S14_L001_R2_001', 'Svenja-TRB-0002-009-BL-INTEGA_S110_L001_R2_001']\n",
      "\n",
      "\n",
      "5-4 C2\n",
      "['NC-hs-TRB-0005-004-2-Zyklus-PB-gDNA_S17_L001_R2_001']\n",
      "\n",
      "\n",
      "7-3 C2\n",
      "['Svenja-TRB-0007-003-BL-INTEGA_S85_L001_R2_001', 'NC-hs-TRB-0007-003-2-Zyklus-PB-gDNA_S19_L001_R2_001']\n",
      "\n",
      "\n",
      "7-4 C2\n",
      "['NC-hs-TRB-0007-004-2-Zyklus-PB-gDNA_S20_L001_R2_001', 'Svenja-TRB-0007-004-BL-INTEGA_S84_L001_R2_001']\n",
      "\n",
      "\n",
      "7-5 C2\n",
      "['NC-hs-TRB-0007-005-2-Zyklus-PB-gDNA_S21_L001_R2_001']\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i, row in double_probe_patient.iterrows(): \n",
    "    id = row.stripped_patient_id\n",
    "    probe = row.Probe_cleaned\n",
    "    print(id, probe)\n",
    "    if probe == \"Baseline\":\n",
    "        probe = [\"Baseline\", \"BL\"]\n",
    "    else:\n",
    "        probe = [probe]\n",
    "    \n",
    "    filename_id = filereport[filereport[\"stripped_patient_id\"] == id][\"filenames\"]\n",
    "    print(list(filename_id))\n",
    "    for file in filename_id:\n",
    "        for p in probe:\n",
    "            if p in file:\n",
    "                print(\"Match: \",file)\n",
    "    print(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "83b5b7f7-bff6-4569-9eae-1d2eb4be2bce",
   "metadata": {},
   "outputs": [],
   "source": [
    "idx_to_drop = [143, 217,183]\n",
    "# overwrite 109 with 110 \n",
    "idx_to_overwrite = {109:110, 48:49, 85:86, 102:103, 140:141, 46:47, 31:32, 42:43, 73:74, 93:94, 96:97, 99:100}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53101923-8f50-4164-9b1c-e69e6c6aab8d",
   "metadata": {},
   "source": [
    "### Overwrite rows with results found in file report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "86c54c41-3f31-485a-b51a-5366ad750340",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in idx_to_overwrite.keys():\n",
    "    j = idx_to_overwrite[i]\n",
    "    Ergebnisse_CDC_Her2.loc[i,'CTC1': 'Probe_cleaned'] = Ergebnisse_CDC_Her2.loc[j,'CTC1': 'Probe_cleaned']\n",
    "    Ergebnisse_CDC_Her2.drop(j,inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51ba38b3-ef97-4acc-b7f4-e54f8e9eff66",
   "metadata": {},
   "source": [
    "### Drop rows with double entry "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "3cf2141b-a93c-4f49-b199-fb932801fb66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Zentrum</th>\n",
       "      <th>Pat-ID</th>\n",
       "      <th>Patient-ID</th>\n",
       "      <th>Probe</th>\n",
       "      <th>NaN</th>\n",
       "      <th>Probenabnahme</th>\n",
       "      <th>Datum_Analyse</th>\n",
       "      <th>CTC1</th>\n",
       "      <th>HER2-0</th>\n",
       "      <th>HER2-1</th>\n",
       "      <th>...</th>\n",
       "      <th>ID</th>\n",
       "      <th>all.names</th>\n",
       "      <th>Clonality</th>\n",
       "      <th>Group</th>\n",
       "      <th>Diversity</th>\n",
       "      <th>Group</th>\n",
       "      <th>Richness</th>\n",
       "      <th>Group</th>\n",
       "      <th>stripped_patient_id</th>\n",
       "      <th>Probe_cleaned</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "      <td>0016-003</td>\n",
       "      <td>1Tag4-5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2019-08-07 00:00:00</td>\n",
       "      <td>2019-08-09 00:00:00</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16-3</td>\n",
       "      <td>1Tag4-5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>38</td>\n",
       "      <td>1</td>\n",
       "      <td>0038-001</td>\n",
       "      <td>1Tag4-5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2019-04-16 00:00:00</td>\n",
       "      <td>2019-04-18 00:00:00</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38-1</td>\n",
       "      <td>1Tag4-5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>0027-002</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019-03-13 00:00:00</td>\n",
       "      <td>2019-03-15 00:00:00</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27-2</td>\n",
       "      <td>BL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Zentrum Pat-ID Patient-ID     Probe  NaN        Probenabnahme  \\\n",
       "143      16      3   0016-003   1Tag4-5  3.0  2019-08-07 00:00:00   \n",
       "217      38      1   0038-001   1Tag4-5  3.0  2019-04-16 00:00:00   \n",
       "183      27      2   0027-002  Baseline  1.0  2019-03-13 00:00:00   \n",
       "\n",
       "           Datum_Analyse CTC1 HER2-0 HER2-1  ...   ID all.names Clonality  \\\n",
       "143  2019-08-09 00:00:00    5      5      0  ...  NaN       NaN       NaN   \n",
       "217  2019-04-18 00:00:00    0    NaN    NaN  ...  NaN       NaN       NaN   \n",
       "183  2019-03-15 00:00:00   10      0      1  ...  NaN       NaN       NaN   \n",
       "\n",
       "    Group Diversity Group Richness Group stripped_patient_id Probe_cleaned  \n",
       "143   NaN       NaN   NaN      NaN   NaN                16-3       1Tag4-5  \n",
       "217   NaN       NaN   NaN      NaN   NaN                38-1       1Tag4-5  \n",
       "183   NaN       NaN   NaN      NaN   NaN                27-2            BL  \n",
       "\n",
       "[3 rows x 31 columns]"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Ergebnisse_CDC_Her2.loc[idx_to_drop]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "1efb9cc1-024f-4585-9e48-5e686f011140",
   "metadata": {},
   "outputs": [],
   "source": [
    "Ergebnisse_CDC_Her2.drop(idx_to_drop, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "id": "58c5b648-9f6f-4c03-92cf-b5c2b7c27cbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "Ergebnisse_CDC_Her2[\"clones.txt.name\"] = Ergebnisse_CDC_Her2[\"all.names\"].apply(lambda x: x + \".clones.txt\" if not pd.isnull(x) else x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "a74b017c-c194-475c-a836-a6744e3a97a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_results = df.merge(Ergebnisse_CDC_Her2, on = [\"clones.txt.name\", \"stripped_patient_id\"], how = \"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "cbc8a751-4d41-4f56-a53b-3fe3fa90815e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "249933"
      ]
     },
     "execution_count": 283,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "d275fefd-2584-4f45-b7e9-63281d109327",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clonality_x</th>\n",
       "      <th>Clonality_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>0.225941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>0.225941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>0.225941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>0.225941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>0.225941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249928</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>0.250719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249929</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>0.250719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249930</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>0.250719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249931</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>0.250719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249932</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>0.250719</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249933 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Clonality_x Clonality_y\n",
       "0       0,211129499744678    0.225941\n",
       "1       0,211129499744678    0.225941\n",
       "2       0,211129499744678    0.225941\n",
       "3       0,211129499744678    0.225941\n",
       "4       0,211129499744678    0.225941\n",
       "...                   ...         ...\n",
       "249928   0,24876120545219    0.250719\n",
       "249929   0,24876120545219    0.250719\n",
       "249930   0,24876120545219    0.250719\n",
       "249931   0,24876120545219    0.250719\n",
       "249932   0,24876120545219    0.250719\n",
       "\n",
       "[249933 rows x 2 columns]"
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_results[[\"Clonality_x\",\"Clonality_y\"]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b714094-2099-4341-99ae-0a069a6c4f9e",
   "metadata": {},
   "source": [
    "## merge proben"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "c0c86fde-290c-4223-aee3-609a7b95a73e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Status</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Tumorproben</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>BL</th>\n",
       "      <th>C2</th>\n",
       "      <th>C5</th>\n",
       "      <th>EOT/PD</th>\n",
       "      <th>Notizen</th>\n",
       "      <th>Cycle_EOT</th>\n",
       "      <th>...</th>\n",
       "      <th>Datum EOT_sample</th>\n",
       "      <th>Datum_EOT_real</th>\n",
       "      <th>week C2 sample</th>\n",
       "      <th>week C5 sample</th>\n",
       "      <th>week EOT sample</th>\n",
       "      <th>week_to_EOT_C2_sample</th>\n",
       "      <th>week_to_EOT_C5_sample</th>\n",
       "      <th>week_to_EOT_EOT_sample</th>\n",
       "      <th>Unnamed: 21</th>\n",
       "      <th>ja: Probe vorhanden</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0001-001</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C16</td>\n",
       "      <td>...</td>\n",
       "      <td>2018-11-29 00:00:00</td>\n",
       "      <td>2018-11-29 00:00:00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>-34.0</td>\n",
       "      <td>-26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA: Probe nicht vorhanden</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0001-002</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C19</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-11-28 00:00:00</td>\n",
       "      <td>2019-11-28 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>-38.0</td>\n",
       "      <td>-32.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>seq: Probe vorhanden und sequenziert mit GeneP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0001-003</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>?</td>\n",
       "      <td>cf</td>\n",
       "      <td>Plasma</td>\n",
       "      <td>seq</td>\n",
       "      <td>BL, C2 und C5 als 0038-001 beschriftet, Patien...</td>\n",
       "      <td>C16</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-10-15 00:00:00</td>\n",
       "      <td>2019-10-15 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>-32.0</td>\n",
       "      <td>-26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cf: cfDNA isoliert, aber noch nicht mit GenePa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0001-004</td>\n",
       "      <td>1 Parafinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C24</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020-09-15 00:00:00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>-51.0</td>\n",
       "      <td>-42.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g: gDNA isoliert, noch kein Gene Panel erstell...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0002-001</td>\n",
       "      <td>1 Parafinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C14</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-10-10 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>-28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Panel: PanelUPCR bereits erstellt, jedoch noch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0038-002</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C31</td>\n",
       "      <td>...</td>\n",
       "      <td>2020-11-03 00:00:00</td>\n",
       "      <td>2020-11-03 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>-61.0</td>\n",
       "      <td>-52.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0038-003</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>cf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>seq</td>\n",
       "      <td>C3 statt C2</td>\n",
       "      <td>C7</td>\n",
       "      <td>...</td>\n",
       "      <td>2020-01-21 00:00:00</td>\n",
       "      <td>2020-01-21 00:00:00</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.0</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0038-004</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>EOT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2</td>\n",
       "      <td>...</td>\n",
       "      <td>2019-12-11 00:00:00</td>\n",
       "      <td>2019-12-11 00:00:00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0038-005</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>Panel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C3</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2020-04-15 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0039-001</td>\n",
       "      <td>1 Paraffinblock</td>\n",
       "      <td>seq</td>\n",
       "      <td>seq</td>\n",
       "      <td>cf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C3</td>\n",
       "      <td>...</td>\n",
       "      <td>2020-01-28 00:00:00</td>\n",
       "      <td>2020-01-28 00:00:00</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>98 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Status   Patient      Tumorproben Tumor   BL     C2      C5 EOT/PD  \\\n",
       "0     NaN  0001-001  1 Paraffinblock   seq  seq    seq     seq    seq   \n",
       "1     NaN  0001-002  1 Paraffinblock   seq  seq    seq     seq    seq   \n",
       "2     NaN  0001-003  1 Paraffinblock   seq    ?     cf  Plasma    seq   \n",
       "3     NaN  0001-004   1 Parafinblock   seq  seq    seq     seq    NaN   \n",
       "4     NaN  0002-001   1 Parafinblock   seq  seq    seq     NaN    NaN   \n",
       "..    ...       ...              ...   ...  ...    ...     ...    ...   \n",
       "93    NaN  0038-002  1 Paraffinblock   seq  seq    seq     seq    seq   \n",
       "94    NaN  0038-003  1 Paraffinblock   seq  seq     cf     NaN    seq   \n",
       "95    NaN  0038-004  1 Paraffinblock   seq  seq    EOT     NaN    seq   \n",
       "96    NaN  0038-005  1 Paraffinblock   seq  seq  Panel     NaN    NaN   \n",
       "97    NaN  0039-001  1 Paraffinblock   seq  seq     cf     NaN    seq   \n",
       "\n",
       "                                              Notizen Cycle_EOT  ...  \\\n",
       "0                                                 NaN       C16  ...   \n",
       "1                                                 NaN       C19  ...   \n",
       "2   BL, C2 und C5 als 0038-001 beschriftet, Patien...       C16  ...   \n",
       "3                                                 NaN       C24  ...   \n",
       "4                                                 NaN       C14  ...   \n",
       "..                                                ...       ...  ...   \n",
       "93                                                NaN       C31  ...   \n",
       "94                                        C3 statt C2        C7  ...   \n",
       "95                                                NaN        C2  ...   \n",
       "96                                                NaN        C3  ...   \n",
       "97                                                NaN        C3  ...   \n",
       "\n",
       "       Datum EOT_sample       Datum_EOT_real week C2 sample week C5 sample  \\\n",
       "0   2018-11-29 00:00:00  2018-11-29 00:00:00            4.0           13.0   \n",
       "1   2019-11-28 00:00:00  2019-11-28 00:00:00            3.0            9.0   \n",
       "2   2019-10-15 00:00:00  2019-10-15 00:00:00            3.0            9.0   \n",
       "3                   NaN  2020-09-15 00:00:00            4.0           13.0   \n",
       "4                   NaN  2018-10-10 00:00:00            3.0            NaN   \n",
       "..                  ...                  ...            ...            ...   \n",
       "93  2020-11-03 00:00:00  2020-11-03 00:00:00            3.0           12.0   \n",
       "94  2020-01-21 00:00:00  2020-01-21 00:00:00            9.0            NaN   \n",
       "95  2019-12-11 00:00:00  2019-12-11 00:00:00            NaN            NaN   \n",
       "96                  NaN  2020-04-15 00:00:00            3.0            NaN   \n",
       "97  2020-01-28 00:00:00  2020-01-28 00:00:00            7.0            NaN   \n",
       "\n",
       "   week EOT sample  week_to_EOT_C2_sample  week_to_EOT_C5_sample  \\\n",
       "0             38.0                  -34.0                  -26.0   \n",
       "1             40.0                  -38.0                  -32.0   \n",
       "2             35.0                  -32.0                  -26.0   \n",
       "3             54.0                  -51.0                  -42.0   \n",
       "4             31.0                  -28.0                    NaN   \n",
       "..             ...                    ...                    ...   \n",
       "93            64.0                  -61.0                  -52.0   \n",
       "94            22.0                  -13.0                    NaN   \n",
       "95             3.0                    NaN                    NaN   \n",
       "96             5.0                   -2.0                    NaN   \n",
       "97            10.0                   -4.0                    NaN   \n",
       "\n",
       "    week_to_EOT_EOT_sample  Unnamed: 21  \\\n",
       "0                      0.0          NaN   \n",
       "1                      0.0          NaN   \n",
       "2                      0.0          NaN   \n",
       "3                      NaN          NaN   \n",
       "4                      NaN          NaN   \n",
       "..                     ...          ...   \n",
       "93                     0.0          NaN   \n",
       "94                     0.0          NaN   \n",
       "95                     0.0          NaN   \n",
       "96                     NaN          NaN   \n",
       "97                     0.0          NaN   \n",
       "\n",
       "                                  ja: Probe vorhanden  \n",
       "0                           NA: Probe nicht vorhanden  \n",
       "1   seq: Probe vorhanden und sequenziert mit GeneP...  \n",
       "2   cf: cfDNA isoliert, aber noch nicht mit GenePa...  \n",
       "3   g: gDNA isoliert, noch kein Gene Panel erstell...  \n",
       "4   Panel: PanelUPCR bereits erstellt, jedoch noch...  \n",
       "..                                                ...  \n",
       "93                                                NaN  \n",
       "94                                                NaN  \n",
       "95                                                NaN  \n",
       "96                                                NaN  \n",
       "97                                                NaN  \n",
       "\n",
       "[98 rows x 23 columns]"
      ]
     },
     "execution_count": 268,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proben_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "2a38f2a9-e0ef-4b72-a33b-0215f7ed906c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([               'Umi.count',           'Umi.proportion',\n",
       "                     'Read.count',          'Read.proportion',\n",
       "       'CDR3.nucleotide.sequence', 'CDR3.amino.acid.sequence',\n",
       "                         'V.gene',                   'J.gene',\n",
       "                         'D.gene',                    'V.end',\n",
       "       ...\n",
       "                             'ID',                'all.names',\n",
       "                    'Clonality_y',                    'Group',\n",
       "                      'Diversity',                    'Group',\n",
       "                       'Richness',                    'Group',\n",
       "          'stripped_patient_id_y',            'Probe_cleaned'],\n",
       "      dtype='object', length=104)"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_results.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "b9154498-8570-4a96-ade8-9f5996cb3821",
   "metadata": {},
   "outputs": [],
   "source": [
    "proben_df[\"stripped_patient_id\"] = proben_df[\"Patient\"].apply(utils.get_stripped_pat_no)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "a98a474f-cd66-4494-9829-dfb9d9607789",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Patients not in disposition:  set()\n",
      "Patients not in twb data:  {'24-1', '15-4', '32-2', '22-3', '9-1', '16-7', '16-4', '1-3', '32-4', '5-5', '6-1', '15-8', '35-4', '33-2', '35-5', '22-5', '2-4'}\n"
     ]
    }
   ],
   "source": [
    "print(\"Patients not in disposition: \", set(df_results[\"stripped_patient_id\"].unique()).difference(set(proben_df[\"stripped_patient_id\"].unique())))\n",
    "print(\"Patients not in twb data: \", set(proben_df[\"stripped_patient_id\"].unique()).difference(set(df_results[\"stripped_patient_id\"].unique())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "540c4b19-4dba-48c3-b7e1-deac221aa72a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_results = df_results.merge(proben_df, on = [\"stripped_patient_id\"], how = \"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "8580038a-d7f9-4824-a3c2-bbade297805d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Umi.count</th>\n",
       "      <th>Umi.proportion</th>\n",
       "      <th>Read.count</th>\n",
       "      <th>Read.proportion</th>\n",
       "      <th>CDR3.nucleotide.sequence</th>\n",
       "      <th>CDR3.amino.acid.sequence</th>\n",
       "      <th>V.gene</th>\n",
       "      <th>J.gene</th>\n",
       "      <th>D.gene</th>\n",
       "      <th>V.end</th>\n",
       "      <th>...</th>\n",
       "      <th>Datum EOT_sample</th>\n",
       "      <th>Datum_EOT_real</th>\n",
       "      <th>week C2 sample</th>\n",
       "      <th>week C5 sample</th>\n",
       "      <th>week EOT sample</th>\n",
       "      <th>week_to_EOT_C2_sample</th>\n",
       "      <th>week_to_EOT_C5_sample</th>\n",
       "      <th>week_to_EOT_EOT_sample</th>\n",
       "      <th>Unnamed: 21</th>\n",
       "      <th>ja: Probe vorhanden</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8221.0</td>\n",
       "      <td>0.208125</td>\n",
       "      <td>8221.0</td>\n",
       "      <td>0.208125</td>\n",
       "      <td>TGCAGCGTTGAAGGTGGGACGCGCAATGAGCAGTTCTTC</td>\n",
       "      <td>CSVEGGTRNEQFF</td>\n",
       "      <td>TRBV29-1</td>\n",
       "      <td>TRBJ2-1</td>\n",
       "      <td>TRBD1, TRBD2</td>\n",
       "      <td>195</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-10-10 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>-28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Panel: PanelUPCR bereits erstellt, jedoch noch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>930.0</td>\n",
       "      <td>0.023522</td>\n",
       "      <td>930.0</td>\n",
       "      <td>0.023522</td>\n",
       "      <td>TGTGCCAAAAATATTGGAGGTGACACCGGGGAGCTGTTTTTT</td>\n",
       "      <td>CAKNIGGDTGELFF</td>\n",
       "      <td>TRBV30</td>\n",
       "      <td>TRBJ2-2</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>177</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-10-10 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>-28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Panel: PanelUPCR bereits erstellt, jedoch noch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>707.0</td>\n",
       "      <td>0.017899</td>\n",
       "      <td>707.0</td>\n",
       "      <td>0.017899</td>\n",
       "      <td>TGCAGCGACCCGACGGTATGCCGGACGACGCCACCGGGGAGCTGTT...</td>\n",
       "      <td>CSDPTVCR~DATGELFF</td>\n",
       "      <td>TRBV29-1</td>\n",
       "      <td>TRBJ2-2</td>\n",
       "      <td></td>\n",
       "      <td>189</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-10-10 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>-28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Panel: PanelUPCR bereits erstellt, jedoch noch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>357.0</td>\n",
       "      <td>0.009036</td>\n",
       "      <td>357.0</td>\n",
       "      <td>0.009036</td>\n",
       "      <td>TGTGCCAGCAGTGAGGGGAGCGGGAGTGGCACAGATACGCAGTATTTT</td>\n",
       "      <td>CASSEGSGSGTDTQYF</td>\n",
       "      <td>TRBV6-1</td>\n",
       "      <td>TRBJ2-3</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>190</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-10-10 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>-28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Panel: PanelUPCR bereits erstellt, jedoch noch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>320.0</td>\n",
       "      <td>0.008078</td>\n",
       "      <td>320.0</td>\n",
       "      <td>0.008078</td>\n",
       "      <td>TGTGCCAGCAGTGAAGGGGCCGGGGCCAACGTCCTGACTTTC</td>\n",
       "      <td>CASSEGAGANVLTF</td>\n",
       "      <td>TRBV6-1</td>\n",
       "      <td>TRBJ2-6</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>192</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-10-10 00:00:00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.0</td>\n",
       "      <td>-28.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Panel: PanelUPCR bereits erstellt, jedoch noch...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 126 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Umi.count  Umi.proportion  Read.count  Read.proportion  \\\n",
       "0     8221.0        0.208125      8221.0         0.208125   \n",
       "1      930.0        0.023522       930.0         0.023522   \n",
       "2      707.0        0.017899       707.0         0.017899   \n",
       "3      357.0        0.009036       357.0         0.009036   \n",
       "4      320.0        0.008078       320.0         0.008078   \n",
       "\n",
       "                            CDR3.nucleotide.sequence CDR3.amino.acid.sequence  \\\n",
       "0            TGCAGCGTTGAAGGTGGGACGCGCAATGAGCAGTTCTTC            CSVEGGTRNEQFF   \n",
       "1         TGTGCCAAAAATATTGGAGGTGACACCGGGGAGCTGTTTTTT           CAKNIGGDTGELFF   \n",
       "2  TGCAGCGACCCGACGGTATGCCGGACGACGCCACCGGGGAGCTGTT...        CSDPTVCR~DATGELFF   \n",
       "3   TGTGCCAGCAGTGAGGGGAGCGGGAGTGGCACAGATACGCAGTATTTT         CASSEGSGSGTDTQYF   \n",
       "4         TGTGCCAGCAGTGAAGGGGCCGGGGCCAACGTCCTGACTTTC           CASSEGAGANVLTF   \n",
       "\n",
       "     V.gene   J.gene        D.gene V.end  ... Datum EOT_sample  \\\n",
       "0  TRBV29-1  TRBJ2-1  TRBD1, TRBD2   195  ...              NaN   \n",
       "1    TRBV30  TRBJ2-2         TRBD2   177  ...              NaN   \n",
       "2  TRBV29-1  TRBJ2-2                 189  ...              NaN   \n",
       "3   TRBV6-1  TRBJ2-3         TRBD2   190  ...              NaN   \n",
       "4   TRBV6-1  TRBJ2-6         TRBD2   192  ...              NaN   \n",
       "\n",
       "        Datum_EOT_real week C2 sample  week C5 sample  week EOT sample  \\\n",
       "0  2018-10-10 00:00:00            3.0             NaN             31.0   \n",
       "1  2018-10-10 00:00:00            3.0             NaN             31.0   \n",
       "2  2018-10-10 00:00:00            3.0             NaN             31.0   \n",
       "3  2018-10-10 00:00:00            3.0             NaN             31.0   \n",
       "4  2018-10-10 00:00:00            3.0             NaN             31.0   \n",
       "\n",
       "   week_to_EOT_C2_sample week_to_EOT_C5_sample  week_to_EOT_EOT_sample  \\\n",
       "0                  -28.0                   NaN                     NaN   \n",
       "1                  -28.0                   NaN                     NaN   \n",
       "2                  -28.0                   NaN                     NaN   \n",
       "3                  -28.0                   NaN                     NaN   \n",
       "4                  -28.0                   NaN                     NaN   \n",
       "\n",
       "  Unnamed: 21                                ja: Probe vorhanden  \n",
       "0         NaN  Panel: PanelUPCR bereits erstellt, jedoch noch...  \n",
       "1         NaN  Panel: PanelUPCR bereits erstellt, jedoch noch...  \n",
       "2         NaN  Panel: PanelUPCR bereits erstellt, jedoch noch...  \n",
       "3         NaN  Panel: PanelUPCR bereits erstellt, jedoch noch...  \n",
       "4         NaN  Panel: PanelUPCR bereits erstellt, jedoch noch...  \n",
       "\n",
       "[5 rows x 126 columns]"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_results.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "28331c70-7e14-4bdd-95b0-e635b26f2f7b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_results.to_pickle(\"intega/intega_40000_reads_with_out_of_frame_merged_dispo_results_proben.pkl\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3551d70a-c849-4a21-be5d-1ebe2c94f305",
   "metadata": {},
   "source": [
    "# Load raw file info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "82a3adb7-f9eb-40bb-a9aa-4818e1ffe150",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "cc2649dea9d24df0b9370828d95a4df0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/132 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_raw = utils.read_clones_txt(np.unique(df[\"clones.txt.name\"].values), \"intega/Intega_final\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "32f917ba-6168-4b8c-829f-f6ab549a6c96",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([               'Umi.count',           'Umi.proportion',\n",
       "                     'Read.count',          'Read.proportion',\n",
       "       'CDR3.nucleotide.sequence', 'CDR3.amino.acid.sequence',\n",
       "                         'V.gene',                   'J.gene',\n",
       "                         'D.gene',                    'V.end',\n",
       "       ...\n",
       "               'Datum EOT_sample',           'Datum_EOT_real',\n",
       "                 'week C2 sample',           'week C5 sample',\n",
       "                'week EOT sample',    'week_to_EOT_C2_sample',\n",
       "          'week_to_EOT_C5_sample',   'week_to_EOT_EOT_sample',\n",
       "                    'Unnamed: 21',      'ja: Probe vorhanden'],\n",
       "      dtype='object', length=126)"
      ]
     },
     "execution_count": 295,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_results.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "5ff1873f-0674-483c-b550-6be6b6e2038c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['cloneId', 'cloneCount', 'cloneFraction', 'nSeqCDR3', 'aaSeqCDR3',\n",
       "       'bestVGene', 'bestDGene', 'bestJGene', 'vBestIdentityPercent',\n",
       "       'lengthOfCDR3', 'targetSequences', 'allVGenes', 'allDGenes',\n",
       "       'allJGenes', 'allVHits', 'allDHits', 'allJHits', 'allVAlignments',\n",
       "       'allDAlignments', 'allJAlignments', 'clones.txt.name'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_raw.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc1eae24-01e3-485e-a388-dff2d483d667",
   "metadata": {},
   "source": [
    "## Add correct clone Id before adding info from raw files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "229008b8-2bbe-40a0-adcb-12851a169ffb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1a8b7d919cd84fcdb5196669b31a48ad",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/132 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for file in tqdm(df_results[\"clones.txt.name\"].unique()):\n",
    "    df_sub_file = df_results[df_results[\"clones.txt.name\"] == file]\n",
    "    df_raw_sub_file = df_raw[df_raw[\"clones.txt.name\"] == file]\n",
    "    for i, index_row in enumerate(df_sub_file.iterrows()):\n",
    "        index, row = index_row[0], index_row[1]  \n",
    "        if row[\"CDR3.nucleotide.sequence\"] == df_raw_sub_file.iloc[i][\"nSeqCDR3\"]:\n",
    "            df_results.loc[index,\"cloneId\"] = df_raw_sub_file.iloc[i][\"cloneId\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "24419d51-4ac3-4560-8d04-965f0fea46cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_results2 = df_results.merge(df_raw, left_on=[\"clones.txt.name\", \"cloneId\", \"CDR3.nucleotide.sequence\"], right_on =[\"clones.txt.name\", \"cloneId\", \"nSeqCDR3\"], how = \"left\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "id": "b6f33254-ca88-4b2f-a754-ab097e1167b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_results2.to_pickle(\"intega/intega_40000_reads_with_out_of_frame_merged_dispo_results_proben_raw_data.pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "id": "c34cfbf8-60fd-45a3-ac87-e20442aa6be3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Umi.count',\n",
       " 'Umi.proportion',\n",
       " 'Read.count',\n",
       " 'Read.proportion',\n",
       " 'CDR3.nucleotide.sequence',\n",
       " 'CDR3.amino.acid.sequence',\n",
       " 'V.gene',\n",
       " 'J.gene',\n",
       " 'D.gene',\n",
       " 'V.end',\n",
       " 'J.start',\n",
       " 'D5.end',\n",
       " 'D3.end',\n",
       " 'VD.insertions',\n",
       " 'DJ.insertions',\n",
       " 'Total.insertions',\n",
       " 'sample',\n",
       " 'cloneId',\n",
       " 'clones.txt.name',\n",
       " '#Nucleotide clones',\n",
       " '#Aminoacid clonotypes',\n",
       " '%Aminoacid clonotypes',\n",
       " '#In-frames',\n",
       " '%In-frames',\n",
       " '#Out-of-frames',\n",
       " '%Out-of-frames',\n",
       " 'Sum.reads',\n",
       " 'Min.reads',\n",
       " '1st Qu.reads',\n",
       " 'Median.reads',\n",
       " 'Mean.reads',\n",
       " '3rd Qu.reads',\n",
       " 'Max.reads',\n",
       " 'Sum.UMIs',\n",
       " 'Min.UMIs',\n",
       " '1st Qu.UMIs',\n",
       " 'Median.UMIs',\n",
       " 'Mean.UMIs',\n",
       " '3rd Qu.UMIs',\n",
       " 'Max.UMIs',\n",
       " 'Clonality_x',\n",
       " 'twb.shannon',\n",
       " 'twb.simp',\n",
       " 'stripped_patient_id',\n",
       " 'Pat-No.',\n",
       " 'Age [y]',\n",
       " 'Gender (M/F)',\n",
       " 'Random No.',\n",
       " 'Treatment as Randomized',\n",
       " 'Treatment as Treated',\n",
       " 'Prior Surgery (Yes/No)',\n",
       " 'HER2 Status Local (IHC 2+/3+)',\n",
       " 'HER2 Status Central',\n",
       " 'Tumor Proportion Score',\n",
       " 'Immune Cell Score',\n",
       " 'Combined Positive Score',\n",
       " 'Prior Drug Therapy',\n",
       " 'Therapy Duration, Any Component [mo]',\n",
       " 'Therapy Duration, All Scheduled Components [mo]',\n",
       " 'Therapy Duration, Nivolumab [mo]',\n",
       " 'Reason for End of Treatment',\n",
       " 'Overall Best Response (CR/PR)',\n",
       " 'Time to Best Response [mo]',\n",
       " 'Response at 1st Tumor Assessment (PD/SD/PR/CR/NA)',\n",
       " 'Sum of Target Lesion Diameters [mm] at Screening',\n",
       " '% Change of Diameters (target lesions) at 1st Tumor Assessment',\n",
       " 'Best % Change of Diameters (Taget Lesions) during Tumor Assessments',\n",
       " 'PFS [mo]',\n",
       " 'Censoring (Yes/No)',\n",
       " 'No of AEs (Grade 3-5)',\n",
       " 'No of SAEs',\n",
       " 'OS [mo]',\n",
       " 'Death (Yes/No)',\n",
       " 'Zentrum',\n",
       " 'Pat-ID',\n",
       " 'Patient-ID',\n",
       " 'Probe',\n",
       " nan,\n",
       " 'Probenabnahme',\n",
       " 'Datum_Analyse',\n",
       " 'CTC1',\n",
       " 'HER2-0',\n",
       " 'HER2-1',\n",
       " 'HER2-2',\n",
       " 'HER2-3',\n",
       " 'CXC',\n",
       " 'PD-L1-0',\n",
       " 'PD-L1-1',\n",
       " 'PD-L1-2',\n",
       " 'PD-L1-3',\n",
       " 'Kommentare1',\n",
       " 'Kommentare2',\n",
       " 'Kommentare3',\n",
       " nan,\n",
       " 'ID',\n",
       " 'all.names',\n",
       " 'Clonality_y',\n",
       " 'Group',\n",
       " 'Diversity',\n",
       " 'Group',\n",
       " 'Richness',\n",
       " 'Group',\n",
       " 'Probe_cleaned',\n",
       " 'Status',\n",
       " 'Patient',\n",
       " 'Tumorproben',\n",
       " 'Tumor',\n",
       " 'BL',\n",
       " 'C2',\n",
       " 'C5',\n",
       " 'EOT/PD',\n",
       " 'Notizen',\n",
       " 'Cycle_EOT',\n",
       " 'Datum BL',\n",
       " 'Datum C2_sample',\n",
       " 'Datum C5_sample',\n",
       " 'Datum EOT_sample',\n",
       " 'Datum_EOT_real',\n",
       " 'week C2 sample',\n",
       " 'week C5 sample',\n",
       " 'week EOT sample',\n",
       " 'week_to_EOT_C2_sample',\n",
       " 'week_to_EOT_C5_sample',\n",
       " 'week_to_EOT_EOT_sample',\n",
       " 'Unnamed: 21',\n",
       " 'ja: Probe vorhanden',\n",
       " 'cloneCount',\n",
       " 'cloneFraction',\n",
       " 'nSeqCDR3',\n",
       " 'aaSeqCDR3',\n",
       " 'bestVGene',\n",
       " 'bestDGene',\n",
       " 'bestJGene',\n",
       " 'vBestIdentityPercent',\n",
       " 'lengthOfCDR3',\n",
       " 'targetSequences',\n",
       " 'allVGenes',\n",
       " 'allDGenes',\n",
       " 'allJGenes',\n",
       " 'allVHits',\n",
       " 'allDHits',\n",
       " 'allJHits',\n",
       " 'allVAlignments',\n",
       " 'allDAlignments',\n",
       " 'allJAlignments']"
      ]
     },
     "execution_count": 307,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(df_results2.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e43ffcd-b66b-4101-ba8c-d4da74f6f59a",
   "metadata": {},
   "source": [
    "# Get rid of redundant information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "id": "59365cc2-b3f5-4ecc-826c-b7607101ade9",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_results2.drop(columns = ['Umi.count', 'Umi.proportion', 'Sum.UMIs', 'Min.UMIs', '1st Qu.UMIs',\n",
    "       'Median.UMIs', 'Mean.UMIs', '3rd Qu.UMIs', 'Max.UMIs', 'nSeqCDR3'],inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "id": "aed3a4fe-03ca-4e49-90e3-191df9b058c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Read.count',\n",
       " 'Read.proportion',\n",
       " 'CDR3.nucleotide.sequence',\n",
       " 'CDR3.amino.acid.sequence',\n",
       " 'V.gene',\n",
       " 'J.gene',\n",
       " 'D.gene',\n",
       " 'V.end',\n",
       " 'J.start',\n",
       " 'D5.end',\n",
       " 'D3.end',\n",
       " 'VD.insertions',\n",
       " 'DJ.insertions',\n",
       " 'Total.insertions',\n",
       " 'sample',\n",
       " 'cloneId',\n",
       " 'clones.txt.name',\n",
       " '#Nucleotide clones',\n",
       " '#Aminoacid clonotypes',\n",
       " '%Aminoacid clonotypes',\n",
       " '#In-frames',\n",
       " '%In-frames',\n",
       " '#Out-of-frames',\n",
       " '%Out-of-frames',\n",
       " 'Sum.reads',\n",
       " 'Min.reads',\n",
       " '1st Qu.reads',\n",
       " 'Median.reads',\n",
       " 'Mean.reads',\n",
       " '3rd Qu.reads',\n",
       " 'Max.reads',\n",
       " 'Clonality_x',\n",
       " 'twb.shannon',\n",
       " 'twb.simp',\n",
       " 'stripped_patient_id',\n",
       " 'Pat-No.',\n",
       " 'Age [y]',\n",
       " 'Gender (M/F)',\n",
       " 'Random No.',\n",
       " 'Treatment as Randomized',\n",
       " 'Treatment as Treated',\n",
       " 'Prior Surgery (Yes/No)',\n",
       " 'HER2 Status Local (IHC 2+/3+)',\n",
       " 'HER2 Status Central',\n",
       " 'Tumor Proportion Score',\n",
       " 'Immune Cell Score',\n",
       " 'Combined Positive Score',\n",
       " 'Prior Drug Therapy',\n",
       " 'Therapy Duration, Any Component [mo]',\n",
       " 'Therapy Duration, All Scheduled Components [mo]',\n",
       " 'Therapy Duration, Nivolumab [mo]',\n",
       " 'Reason for End of Treatment',\n",
       " 'Overall Best Response (CR/PR)',\n",
       " 'Time to Best Response [mo]',\n",
       " 'Response at 1st Tumor Assessment (PD/SD/PR/CR/NA)',\n",
       " 'Sum of Target Lesion Diameters [mm] at Screening',\n",
       " '% Change of Diameters (target lesions) at 1st Tumor Assessment',\n",
       " 'Best % Change of Diameters (Taget Lesions) during Tumor Assessments',\n",
       " 'PFS [mo]',\n",
       " 'Censoring (Yes/No)',\n",
       " 'No of AEs (Grade 3-5)',\n",
       " 'No of SAEs',\n",
       " 'OS [mo]',\n",
       " 'Death (Yes/No)',\n",
       " 'Zentrum',\n",
       " 'Pat-ID',\n",
       " 'Patient-ID',\n",
       " 'Probe',\n",
       " nan,\n",
       " 'Probenabnahme',\n",
       " 'Datum_Analyse',\n",
       " 'CTC1',\n",
       " 'HER2-0',\n",
       " 'HER2-1',\n",
       " 'HER2-2',\n",
       " 'HER2-3',\n",
       " 'CXC',\n",
       " 'PD-L1-0',\n",
       " 'PD-L1-1',\n",
       " 'PD-L1-2',\n",
       " 'PD-L1-3',\n",
       " 'Kommentare1',\n",
       " 'Kommentare2',\n",
       " 'Kommentare3',\n",
       " nan,\n",
       " 'ID',\n",
       " 'all.names',\n",
       " 'Clonality_y',\n",
       " 'Group',\n",
       " 'Diversity',\n",
       " 'Group',\n",
       " 'Richness',\n",
       " 'Group',\n",
       " 'Probe_cleaned',\n",
       " 'Status',\n",
       " 'Patient',\n",
       " 'Tumorproben',\n",
       " 'Tumor',\n",
       " 'BL',\n",
       " 'C2',\n",
       " 'C5',\n",
       " 'EOT/PD',\n",
       " 'Notizen',\n",
       " 'Cycle_EOT',\n",
       " 'Datum BL',\n",
       " 'Datum C2_sample',\n",
       " 'Datum C5_sample',\n",
       " 'Datum EOT_sample',\n",
       " 'Datum_EOT_real',\n",
       " 'week C2 sample',\n",
       " 'week C5 sample',\n",
       " 'week EOT sample',\n",
       " 'week_to_EOT_C2_sample',\n",
       " 'week_to_EOT_C5_sample',\n",
       " 'week_to_EOT_EOT_sample',\n",
       " 'Unnamed: 21',\n",
       " 'ja: Probe vorhanden',\n",
       " 'cloneCount',\n",
       " 'cloneFraction',\n",
       " 'aaSeqCDR3',\n",
       " 'bestVGene',\n",
       " 'bestDGene',\n",
       " 'bestJGene',\n",
       " 'vBestIdentityPercent',\n",
       " 'lengthOfCDR3',\n",
       " 'targetSequences',\n",
       " 'allVGenes',\n",
       " 'allDGenes',\n",
       " 'allJGenes',\n",
       " 'allVHits',\n",
       " 'allDHits',\n",
       " 'allJHits',\n",
       " 'allVAlignments',\n",
       " 'allDAlignments',\n",
       " 'allJAlignments']"
      ]
     },
     "execution_count": 311,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(df_results2.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "ffc86590-8f60-46d5-acaf-dc27be0c5f72",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pat-No.</th>\n",
       "      <th>Pat-ID</th>\n",
       "      <th>Patient-ID</th>\n",
       "      <th>Probe</th>\n",
       "      <th>ID</th>\n",
       "      <th>all.names</th>\n",
       "      <th>Group</th>\n",
       "      <th>Group</th>\n",
       "      <th>Group</th>\n",
       "      <th>Probe_cleaned</th>\n",
       "      <th>Patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0002-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0002-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>002-001</td>\n",
       "      <td>Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0002-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0002-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0002-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>002-001</td>\n",
       "      <td>Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0002-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0002-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0002-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>002-001</td>\n",
       "      <td>Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0002-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0002-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0002-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>002-001</td>\n",
       "      <td>Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0002-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0002-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0002-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>002-001</td>\n",
       "      <td>Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0002-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249928</th>\n",
       "      <td>0035-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Svenja-TRB-0035-001-BL-INTEGA_S145</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0035-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249929</th>\n",
       "      <td>0035-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Svenja-TRB-0035-001-BL-INTEGA_S145</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0035-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249930</th>\n",
       "      <td>0035-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Svenja-TRB-0035-001-BL-INTEGA_S145</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0035-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249931</th>\n",
       "      <td>0035-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Svenja-TRB-0035-001-BL-INTEGA_S145</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0035-001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249932</th>\n",
       "      <td>0035-001</td>\n",
       "      <td>1</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Baseline</td>\n",
       "      <td>0035-001</td>\n",
       "      <td>Svenja-TRB-0035-001-BL-INTEGA_S145</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>BL</td>\n",
       "      <td>0035-001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249933 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Pat-No. Pat-ID Patient-ID     Probe        ID  \\\n",
       "0       0002-001      1   0002-001  Baseline   002-001   \n",
       "1       0002-001      1   0002-001  Baseline   002-001   \n",
       "2       0002-001      1   0002-001  Baseline   002-001   \n",
       "3       0002-001      1   0002-001  Baseline   002-001   \n",
       "4       0002-001      1   0002-001  Baseline   002-001   \n",
       "...          ...    ...        ...       ...       ...   \n",
       "249928  0035-001      1   0035-001  Baseline  0035-001   \n",
       "249929  0035-001      1   0035-001  Baseline  0035-001   \n",
       "249930  0035-001      1   0035-001  Baseline  0035-001   \n",
       "249931  0035-001      1   0035-001  Baseline  0035-001   \n",
       "249932  0035-001      1   0035-001  Baseline  0035-001   \n",
       "\n",
       "                                        all.names Group Group Group  \\\n",
       "0       Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111    BL    BL    BL   \n",
       "1       Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111    BL    BL    BL   \n",
       "2       Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111    BL    BL    BL   \n",
       "3       Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111    BL    BL    BL   \n",
       "4       Dona-hs-TRB-002-001-PB-Baseline-gDNA_S111    BL    BL    BL   \n",
       "...                                           ...   ...   ...   ...   \n",
       "249928         Svenja-TRB-0035-001-BL-INTEGA_S145    BL    BL    BL   \n",
       "249929         Svenja-TRB-0035-001-BL-INTEGA_S145    BL    BL    BL   \n",
       "249930         Svenja-TRB-0035-001-BL-INTEGA_S145    BL    BL    BL   \n",
       "249931         Svenja-TRB-0035-001-BL-INTEGA_S145    BL    BL    BL   \n",
       "249932         Svenja-TRB-0035-001-BL-INTEGA_S145    BL    BL    BL   \n",
       "\n",
       "       Probe_cleaned   Patient  \n",
       "0                 BL  0002-001  \n",
       "1                 BL  0002-001  \n",
       "2                 BL  0002-001  \n",
       "3                 BL  0002-001  \n",
       "4                 BL  0002-001  \n",
       "...              ...       ...  \n",
       "249928            BL  0035-001  \n",
       "249929            BL  0035-001  \n",
       "249930            BL  0035-001  \n",
       "249931            BL  0035-001  \n",
       "249932            BL  0035-001  \n",
       "\n",
       "[249933 rows x 11 columns]"
      ]
     },
     "execution_count": 315,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_results2[['Pat-No.','Pat-ID','Patient-ID', 'Probe', 'ID', 'all.names', 'Group',\n",
    " 'Probe_cleaned', 'Patient']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "faf627c4-a4cc-4d65-8226-94a3a2af99a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_results2.drop(columns = ['Pat-No.','Pat-ID','Patient-ID','ID', 'all.names', 'Group','Patient'],inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 317,
   "id": "de711f44-743c-46c5-8094-5a9ca217c500",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clonality_x</th>\n",
       "      <th>twb.shannon</th>\n",
       "      <th>twb.simp</th>\n",
       "      <th>Clonality_y</th>\n",
       "      <th>Diversity</th>\n",
       "      <th>Richness</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>8,22992454789389</td>\n",
       "      <td>22,0566329934599</td>\n",
       "      <td>0.225941</td>\n",
       "      <td>7.659172</td>\n",
       "      <td>952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>8,22992454789389</td>\n",
       "      <td>22,0566329934599</td>\n",
       "      <td>0.225941</td>\n",
       "      <td>7.659172</td>\n",
       "      <td>952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>8,22992454789389</td>\n",
       "      <td>22,0566329934599</td>\n",
       "      <td>0.225941</td>\n",
       "      <td>7.659172</td>\n",
       "      <td>952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>8,22992454789389</td>\n",
       "      <td>22,0566329934599</td>\n",
       "      <td>0.225941</td>\n",
       "      <td>7.659172</td>\n",
       "      <td>952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0,211129499744678</td>\n",
       "      <td>8,22992454789389</td>\n",
       "      <td>22,0566329934599</td>\n",
       "      <td>0.225941</td>\n",
       "      <td>7.659172</td>\n",
       "      <td>952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249928</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>8,00753241247974</td>\n",
       "      <td>46,2592236125888</td>\n",
       "      <td>0.250719</td>\n",
       "      <td>7.448633</td>\n",
       "      <td>983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249929</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>8,00753241247974</td>\n",
       "      <td>46,2592236125888</td>\n",
       "      <td>0.250719</td>\n",
       "      <td>7.448633</td>\n",
       "      <td>983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249930</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>8,00753241247974</td>\n",
       "      <td>46,2592236125888</td>\n",
       "      <td>0.250719</td>\n",
       "      <td>7.448633</td>\n",
       "      <td>983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249931</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>8,00753241247974</td>\n",
       "      <td>46,2592236125888</td>\n",
       "      <td>0.250719</td>\n",
       "      <td>7.448633</td>\n",
       "      <td>983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249932</th>\n",
       "      <td>0,24876120545219</td>\n",
       "      <td>8,00753241247974</td>\n",
       "      <td>46,2592236125888</td>\n",
       "      <td>0.250719</td>\n",
       "      <td>7.448633</td>\n",
       "      <td>983</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249933 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Clonality_x       twb.shannon          twb.simp Clonality_y  \\\n",
       "0       0,211129499744678  8,22992454789389  22,0566329934599    0.225941   \n",
       "1       0,211129499744678  8,22992454789389  22,0566329934599    0.225941   \n",
       "2       0,211129499744678  8,22992454789389  22,0566329934599    0.225941   \n",
       "3       0,211129499744678  8,22992454789389  22,0566329934599    0.225941   \n",
       "4       0,211129499744678  8,22992454789389  22,0566329934599    0.225941   \n",
       "...                   ...               ...               ...         ...   \n",
       "249928   0,24876120545219  8,00753241247974  46,2592236125888    0.250719   \n",
       "249929   0,24876120545219  8,00753241247974  46,2592236125888    0.250719   \n",
       "249930   0,24876120545219  8,00753241247974  46,2592236125888    0.250719   \n",
       "249931   0,24876120545219  8,00753241247974  46,2592236125888    0.250719   \n",
       "249932   0,24876120545219  8,00753241247974  46,2592236125888    0.250719   \n",
       "\n",
       "       Diversity Richness  \n",
       "0       7.659172      952  \n",
       "1       7.659172      952  \n",
       "2       7.659172      952  \n",
       "3       7.659172      952  \n",
       "4       7.659172      952  \n",
       "...          ...      ...  \n",
       "249928  7.448633      983  \n",
       "249929  7.448633      983  \n",
       "249930  7.448633      983  \n",
       "249931  7.448633      983  \n",
       "249932  7.448633      983  \n",
       "\n",
       "[249933 rows x 6 columns]"
      ]
     },
     "execution_count": 317,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_results2[['Clonality_x',\n",
    " 'twb.shannon',\n",
    " 'twb.simp','Clonality_y','Diversity', 'Richness',]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "dbfb7e9e-e3d8-4af5-ba3c-1abf953ed7b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_results2.rename(columns={\"Clonality_x\" : \"Clonality\", \n",
    "                            \"Clonality_y\" : \"Clonality_results\",\n",
    "                           \"Diversity\" : \"Diversity_results\", \n",
    "                           \"Richness\" : \"Richness_results\"}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "642dba83-1d82-4b8e-9579-a221e9a99613",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_results2.to_pickle(\"intega/intega_40000_reads_with_out_of_frame_merged_dispo_results_proben_raw_data_cleaned.pkl\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06e82a9b-6c0c-40cd-ade7-e673199ee69d",
   "metadata": {},
   "source": [
    "# Small Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fa352e93-71f3-4959-9bc4-fbcec7b6f177",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_pickle(\"intega/intega_40000_reads_with_out_of_frame_merged_dispo_results_proben_raw_data_cleaned.pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4230681f-bc8e-45b2-bfcf-e82250c3cdcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Read.count', 'Read.proportion', 'CDR3.nucleotide.sequence',\n",
       "       'CDR3.amino.acid.sequence', 'V.gene', 'J.gene', 'D.gene', 'V.end',\n",
       "       'J.start', 'D5.end', 'D3.end', 'VD.insertions', 'DJ.insertions',\n",
       "       'Total.insertions', 'sample', 'cloneId', 'clones.txt.name',\n",
       "       '#Nucleotide clones', '#Aminoacid clonotypes',\n",
       "       '%Aminoacid clonotypes', '#In-frames', '%In-frames',\n",
       "       '#Out-of-frames', '%Out-of-frames', 'Sum.reads', 'Min.reads',\n",
       "       '1st Qu.reads', 'Median.reads', 'Mean.reads', '3rd Qu.reads',\n",
       "       'Max.reads', 'Clonality', 'twb.shannon', 'twb.simp',\n",
       "       'stripped_patient_id', 'Age [y]', 'Gender (M/F)', 'Random No.',\n",
       "       'Treatment as Randomized', 'Treatment as Treated',\n",
       "       'Prior Surgery (Yes/No)', 'HER2 Status Local (IHC 2+/3+)',\n",
       "       'HER2 Status Central', 'Tumor Proportion Score',\n",
       "       'Immune Cell Score', 'Combined Positive Score',\n",
       "       'Prior Drug Therapy', 'Therapy Duration, Any Component [mo]',\n",
       "       'Therapy Duration, All Scheduled Components [mo]',\n",
       "       'Therapy Duration, Nivolumab [mo]', 'Reason for End of Treatment',\n",
       "       'Overall Best Response (CR/PR)', 'Time to Best Response [mo]',\n",
       "       'Response at 1st Tumor Assessment (PD/SD/PR/CR/NA)',\n",
       "       'Sum of Target Lesion Diameters [mm] at Screening',\n",
       "       '% Change of Diameters (target lesions) at 1st Tumor Assessment',\n",
       "       'Best % Change of Diameters (Taget Lesions) during Tumor Assessments',\n",
       "       'PFS [mo]', 'Censoring (Yes/No)', 'No of AEs (Grade 3-5)',\n",
       "       'No of SAEs', 'OS [mo]', 'Death (Yes/No)', 'Zentrum', 'Probe', nan,\n",
       "       'Probenabnahme', 'Datum_Analyse', 'CTC1', 'HER2-0', 'HER2-1',\n",
       "       'HER2-2', 'HER2-3', 'CXC', 'PD-L1-0', 'PD-L1-1', 'PD-L1-2',\n",
       "       'PD-L1-3', 'Kommentare1', 'Kommentare2', 'Kommentare3', nan,\n",
       "       'Clonality_results', 'Diversity_results', 'Richness_results',\n",
       "       'Probe_cleaned', 'Status', 'Tumorproben', 'Tumor', 'BL', 'C2',\n",
       "       'C5', 'EOT/PD', 'Notizen', 'Cycle_EOT', 'Datum BL',\n",
       "       'Datum C2_sample', 'Datum C5_sample', 'Datum EOT_sample',\n",
       "       'Datum_EOT_real', 'week C2 sample', 'week C5 sample',\n",
       "       'week EOT sample', 'week_to_EOT_C2_sample',\n",
       "       'week_to_EOT_C5_sample', 'week_to_EOT_EOT_sample', 'Unnamed: 21',\n",
       "       'ja: Probe vorhanden', 'cloneCount', 'cloneFraction', 'aaSeqCDR3',\n",
       "       'bestVGene', 'bestDGene', 'bestJGene', 'vBestIdentityPercent',\n",
       "       'lengthOfCDR3', 'targetSequences', 'allVGenes', 'allDGenes',\n",
       "       'allJGenes', 'allVHits', 'allDHits', 'allJHits', 'allVAlignments',\n",
       "       'allDAlignments', 'allJAlignments'], dtype=object)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.asarray(df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "88e5e080-05eb-4e9c-b7e9-dc374a6e0020",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Read.count</th>\n",
       "      <th>Read.proportion</th>\n",
       "      <th>CDR3.nucleotide.sequence</th>\n",
       "      <th>CDR3.amino.acid.sequence</th>\n",
       "      <th>V.gene</th>\n",
       "      <th>J.gene</th>\n",
       "      <th>D.gene</th>\n",
       "      <th>V.end</th>\n",
       "      <th>J.start</th>\n",
       "      <th>D5.end</th>\n",
       "      <th>...</th>\n",
       "      <th>targetSequences</th>\n",
       "      <th>allVGenes</th>\n",
       "      <th>allDGenes</th>\n",
       "      <th>allJGenes</th>\n",
       "      <th>allVHits</th>\n",
       "      <th>allDHits</th>\n",
       "      <th>allJHits</th>\n",
       "      <th>allVAlignments</th>\n",
       "      <th>allDAlignments</th>\n",
       "      <th>allJAlignments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8221.0</td>\n",
       "      <td>0.208125</td>\n",
       "      <td>TGCAGCGTTGAAGGTGGGACGCGCAATGAGCAGTTCTTC</td>\n",
       "      <td>CSVEGGTRNEQFF</td>\n",
       "      <td>TRBV29-1</td>\n",
       "      <td>TRBJ2-1</td>\n",
       "      <td>TRBD1, TRBD2</td>\n",
       "      <td>195</td>\n",
       "      <td>205</td>\n",
       "      <td>197</td>\n",
       "      <td>...</td>\n",
       "      <td>TGTACTGGTACCGTCAGCAACCTGGACAGAGCCTGACACTGATCGC...</td>\n",
       "      <td>TRBV29-1</td>\n",
       "      <td>TRBD2,TRBD1</td>\n",
       "      <td>TRBJ2-1</td>\n",
       "      <td>TRBV29-1*00</td>\n",
       "      <td>TRBD1*00,TRBD2*00</td>\n",
       "      <td>TRBJ2-1*00</td>\n",
       "      <td>94|289|310|0|195|ST97A|961.0</td>\n",
       "      <td>12|17|36|197|202||25.0;16|21|48|197|202||25.0</td>\n",
       "      <td>26|70|70|205|249||220.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>930.0</td>\n",
       "      <td>0.023522</td>\n",
       "      <td>TGTGCCAAAAATATTGGAGGTGACACCGGGGAGCTGTTTTTT</td>\n",
       "      <td>CAKNIGGDTGELFF</td>\n",
       "      <td>TRBV30</td>\n",
       "      <td>TRBJ2-2</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>177</td>\n",
       "      <td>193</td>\n",
       "      <td>186</td>\n",
       "      <td>...</td>\n",
       "      <td>TACTGGTACCGACAGGCTGCAGGCAGGGGCCTCCAGCTGCTCTTCT...</td>\n",
       "      <td>TRBV30</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>TRBJ2-2</td>\n",
       "      <td>TRBV30*00</td>\n",
       "      <td>TRBD2*00</td>\n",
       "      <td>TRBJ2-2*00</td>\n",
       "      <td>99|276|304|0|177||885.0</td>\n",
       "      <td>26|31|48|186|191||25.0</td>\n",
       "      <td>23|71|71|193|241||240.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>707.0</td>\n",
       "      <td>0.017899</td>\n",
       "      <td>TGCAGCGACCCGACGGTATGCCGGACGACGCCACCGGGGAGCTGTT...</td>\n",
       "      <td>CSDPTVCR~DATGELFF</td>\n",
       "      <td>TRBV29-1</td>\n",
       "      <td>TRBJ2-2</td>\n",
       "      <td></td>\n",
       "      <td>189</td>\n",
       "      <td>213</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>TGTACTGGTACCGTCAGCAACCTGGACAGAGCCTGACACTGATCGC...</td>\n",
       "      <td>TRBV29-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ2-2</td>\n",
       "      <td>TRBV29-1*00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ2-2*00</td>\n",
       "      <td>94|283|310|0|189|ST97A|931.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24|71|71|213|260||235.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>357.0</td>\n",
       "      <td>0.009036</td>\n",
       "      <td>TGTGCCAGCAGTGAGGGGAGCGGGAGTGGCACAGATACGCAGTATTTT</td>\n",
       "      <td>CASSEGSGSGTDTQYF</td>\n",
       "      <td>TRBV6-1</td>\n",
       "      <td>TRBJ2-3</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>190</td>\n",
       "      <td>204</td>\n",
       "      <td>194</td>\n",
       "      <td>...</td>\n",
       "      <td>TGTACTGGTATCGACAAGACCCAGGCATGGGACTGAGGCTGATTTA...</td>\n",
       "      <td>TRBV6-1</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>TRBJ2-3</td>\n",
       "      <td>TRBV6-1*00</td>\n",
       "      <td>TRBD2*00</td>\n",
       "      <td>TRBJ2-3*00</td>\n",
       "      <td>94|284|307|0|190||950.0</td>\n",
       "      <td>22|30|48|194|202||40.0</td>\n",
       "      <td>21|69|69|204|252|SA61CSC67G|212.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>320.0</td>\n",
       "      <td>0.008078</td>\n",
       "      <td>TGTGCCAGCAGTGAAGGGGCCGGGGCCAACGTCCTGACTTTC</td>\n",
       "      <td>CASSEGAGANVLTF</td>\n",
       "      <td>TRBV6-1</td>\n",
       "      <td>TRBJ2-6</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>192</td>\n",
       "      <td>197</td>\n",
       "      <td>192</td>\n",
       "      <td>...</td>\n",
       "      <td>TGTACTGGTATCGACAAGACCCAGGCATGGGACTGAGGCTGATTTA...</td>\n",
       "      <td>TRBV6-1</td>\n",
       "      <td>TRBD2</td>\n",
       "      <td>TRBJ2-6</td>\n",
       "      <td>TRBV6-1*00</td>\n",
       "      <td>TRBD2*00</td>\n",
       "      <td>TRBJ2-6*00</td>\n",
       "      <td>94|286|307|0|192||960.0</td>\n",
       "      <td>29|34|48|192|197||25.0</td>\n",
       "      <td>24|73|73|197|246||245.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249928</th>\n",
       "      <td>3.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>TGCAGTGCTGGGGCCTCAGATACGCAGTATTTT</td>\n",
       "      <td>CSAGASDTQYF</td>\n",
       "      <td>TRBV20-1</td>\n",
       "      <td>TRBJ2-3</td>\n",
       "      <td>TRBD1, TRBD2</td>\n",
       "      <td>194</td>\n",
       "      <td>201</td>\n",
       "      <td>194</td>\n",
       "      <td>...</td>\n",
       "      <td>ATGTTTACTGGTATCGTCAGTTCCCGAAACAGAGTCTCATGCTGAT...</td>\n",
       "      <td>TRBV20-1</td>\n",
       "      <td>TRBD2,TRBD1</td>\n",
       "      <td>TRBJ2-3</td>\n",
       "      <td>TRBV20-1*00</td>\n",
       "      <td>TRBD1*00,TRBD2*00</td>\n",
       "      <td>TRBJ2-3*00</td>\n",
       "      <td>96|288|313|0|194|I102AI102C|936.0</td>\n",
       "      <td>19|24|36|194|199||25.0;29|34|48|195|200||25.0</td>\n",
       "      <td>24|69|69|201|246|SC67G|211.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249929</th>\n",
       "      <td>3.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>TGCGCCAGCTCGAAAACTGTTTTTT</td>\n",
       "      <td>CASSK~KLFF</td>\n",
       "      <td>TRBV10-2</td>\n",
       "      <td>TRBJ1-4</td>\n",
       "      <td></td>\n",
       "      <td>185</td>\n",
       "      <td>188</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>TGTACTGGTATCGACAAGACCTGGGACATGGGCTGAGGCTGATCTA...</td>\n",
       "      <td>TRBV10-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ1-4</td>\n",
       "      <td>TRBV10-2*00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ1-4*00</td>\n",
       "      <td>94|279|307|0|185|ST97A|911.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30|71|71|188|229||205.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249930</th>\n",
       "      <td>3.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>TGTGCCAGCAGCTTAGTGGTTGAAGGAGAGACCCAGTACTTC</td>\n",
       "      <td>CASSLVVEGETQYF</td>\n",
       "      <td>TRBV7-9</td>\n",
       "      <td>TRBJ2-5</td>\n",
       "      <td></td>\n",
       "      <td>195</td>\n",
       "      <td>205</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>TTTATTGGTACCGACAGACCCTGGGGCAGGGCCCAGAGTTTCTGAC...</td>\n",
       "      <td>TRBV7-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ2-5</td>\n",
       "      <td>TRBV7-9*00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ2-5*00</td>\n",
       "      <td>94|289|310|0|195||975.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24|68|68|205|249|SC66G|206.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249931</th>\n",
       "      <td>3.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>TGCAGTGCCCCCCTGTATCAGGGTCCCGAGCAGTACTTC</td>\n",
       "      <td>CSAPLYQGPEQYF</td>\n",
       "      <td>TRBV20-1</td>\n",
       "      <td>TRBJ2-7</td>\n",
       "      <td>TRBD1</td>\n",
       "      <td>193</td>\n",
       "      <td>211</td>\n",
       "      <td>193</td>\n",
       "      <td>...</td>\n",
       "      <td>CTATGTTTTGGTATCGTCAGTTCCCGAAACAGAGTCTCATGCTGAT...</td>\n",
       "      <td>TRBV20-1</td>\n",
       "      <td>TRBD1</td>\n",
       "      <td>TRBJ2-7</td>\n",
       "      <td>TRBV20-1*00</td>\n",
       "      <td>TRBD1*00</td>\n",
       "      <td>TRBJ2-7*00</td>\n",
       "      <td>94|287|313|0|193||965.0</td>\n",
       "      <td>1|9|36|193|201||40.0</td>\n",
       "      <td>26|59|67|211|244||165.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249932</th>\n",
       "      <td>3.0</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>TGTGCCAGCAGCTTAGATGCCAGCCGAGGAAAACTGTTTTTT</td>\n",
       "      <td>CASSLDASRGKLFF</td>\n",
       "      <td>TRBV11-2</td>\n",
       "      <td>TRBJ1-4</td>\n",
       "      <td></td>\n",
       "      <td>201</td>\n",
       "      <td>212</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>ACCCTTTACTGGTACCGGCAGATCCTGGGACAGGGCCCAAAGCTTC...</td>\n",
       "      <td>TRBV11-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ1-4</td>\n",
       "      <td>TRBV11-2*00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRBJ1-4*00</td>\n",
       "      <td>90|291|310|0|201|SA106G|991.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30|71|71|212|253||205.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249933 rows × 126 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Read.count  Read.proportion  \\\n",
       "0           8221.0         0.208125   \n",
       "1            930.0         0.023522   \n",
       "2            707.0         0.017899   \n",
       "3            357.0         0.009036   \n",
       "4            320.0         0.008078   \n",
       "...            ...              ...   \n",
       "249928         3.0         0.000056   \n",
       "249929         3.0         0.000056   \n",
       "249930         3.0         0.000056   \n",
       "249931         3.0         0.000056   \n",
       "249932         3.0         0.000056   \n",
       "\n",
       "                                 CDR3.nucleotide.sequence  \\\n",
       "0                 TGCAGCGTTGAAGGTGGGACGCGCAATGAGCAGTTCTTC   \n",
       "1              TGTGCCAAAAATATTGGAGGTGACACCGGGGAGCTGTTTTTT   \n",
       "2       TGCAGCGACCCGACGGTATGCCGGACGACGCCACCGGGGAGCTGTT...   \n",
       "3        TGTGCCAGCAGTGAGGGGAGCGGGAGTGGCACAGATACGCAGTATTTT   \n",
       "4              TGTGCCAGCAGTGAAGGGGCCGGGGCCAACGTCCTGACTTTC   \n",
       "...                                                   ...   \n",
       "249928                  TGCAGTGCTGGGGCCTCAGATACGCAGTATTTT   \n",
       "249929                          TGCGCCAGCTCGAAAACTGTTTTTT   \n",
       "249930         TGTGCCAGCAGCTTAGTGGTTGAAGGAGAGACCCAGTACTTC   \n",
       "249931            TGCAGTGCCCCCCTGTATCAGGGTCCCGAGCAGTACTTC   \n",
       "249932         TGTGCCAGCAGCTTAGATGCCAGCCGAGGAAAACTGTTTTTT   \n",
       "\n",
       "       CDR3.amino.acid.sequence    V.gene   J.gene        D.gene V.end  \\\n",
       "0                 CSVEGGTRNEQFF  TRBV29-1  TRBJ2-1  TRBD1, TRBD2   195   \n",
       "1                CAKNIGGDTGELFF    TRBV30  TRBJ2-2         TRBD2   177   \n",
       "2             CSDPTVCR~DATGELFF  TRBV29-1  TRBJ2-2                 189   \n",
       "3              CASSEGSGSGTDTQYF   TRBV6-1  TRBJ2-3         TRBD2   190   \n",
       "4                CASSEGAGANVLTF   TRBV6-1  TRBJ2-6         TRBD2   192   \n",
       "...                         ...       ...      ...           ...   ...   \n",
       "249928              CSAGASDTQYF  TRBV20-1  TRBJ2-3  TRBD1, TRBD2   194   \n",
       "249929               CASSK~KLFF  TRBV10-2  TRBJ1-4                 185   \n",
       "249930           CASSLVVEGETQYF   TRBV7-9  TRBJ2-5                 195   \n",
       "249931            CSAPLYQGPEQYF  TRBV20-1  TRBJ2-7         TRBD1   193   \n",
       "249932           CASSLDASRGKLFF  TRBV11-2  TRBJ1-4                 201   \n",
       "\n",
       "       J.start D5.end  ...                                    targetSequences  \\\n",
       "0          205    197  ...  TGTACTGGTACCGTCAGCAACCTGGACAGAGCCTGACACTGATCGC...   \n",
       "1          193    186  ...  TACTGGTACCGACAGGCTGCAGGCAGGGGCCTCCAGCTGCTCTTCT...   \n",
       "2          213     -1  ...  TGTACTGGTACCGTCAGCAACCTGGACAGAGCCTGACACTGATCGC...   \n",
       "3          204    194  ...  TGTACTGGTATCGACAAGACCCAGGCATGGGACTGAGGCTGATTTA...   \n",
       "4          197    192  ...  TGTACTGGTATCGACAAGACCCAGGCATGGGACTGAGGCTGATTTA...   \n",
       "...        ...    ...  ...                                                ...   \n",
       "249928     201    194  ...  ATGTTTACTGGTATCGTCAGTTCCCGAAACAGAGTCTCATGCTGAT...   \n",
       "249929     188     -1  ...  TGTACTGGTATCGACAAGACCTGGGACATGGGCTGAGGCTGATCTA...   \n",
       "249930     205     -1  ...  TTTATTGGTACCGACAGACCCTGGGGCAGGGCCCAGAGTTTCTGAC...   \n",
       "249931     211    193  ...  CTATGTTTTGGTATCGTCAGTTCCCGAAACAGAGTCTCATGCTGAT...   \n",
       "249932     212     -1  ...  ACCCTTTACTGGTACCGGCAGATCCTGGGACAGGGCCCAAAGCTTC...   \n",
       "\n",
       "        allVGenes    allDGenes  allJGenes     allVHits           allDHits  \\\n",
       "0        TRBV29-1  TRBD2,TRBD1    TRBJ2-1  TRBV29-1*00  TRBD1*00,TRBD2*00   \n",
       "1          TRBV30        TRBD2    TRBJ2-2    TRBV30*00           TRBD2*00   \n",
       "2        TRBV29-1          NaN    TRBJ2-2  TRBV29-1*00                NaN   \n",
       "3         TRBV6-1        TRBD2    TRBJ2-3   TRBV6-1*00           TRBD2*00   \n",
       "4         TRBV6-1        TRBD2    TRBJ2-6   TRBV6-1*00           TRBD2*00   \n",
       "...           ...          ...        ...          ...                ...   \n",
       "249928   TRBV20-1  TRBD2,TRBD1    TRBJ2-3  TRBV20-1*00  TRBD1*00,TRBD2*00   \n",
       "249929   TRBV10-2          NaN    TRBJ1-4  TRBV10-2*00                NaN   \n",
       "249930    TRBV7-9          NaN    TRBJ2-5   TRBV7-9*00                NaN   \n",
       "249931   TRBV20-1        TRBD1    TRBJ2-7  TRBV20-1*00           TRBD1*00   \n",
       "249932   TRBV11-2          NaN    TRBJ1-4  TRBV11-2*00                NaN   \n",
       "\n",
       "          allJHits                     allVAlignments  \\\n",
       "0       TRBJ2-1*00       94|289|310|0|195|ST97A|961.0   \n",
       "1       TRBJ2-2*00            99|276|304|0|177||885.0   \n",
       "2       TRBJ2-2*00       94|283|310|0|189|ST97A|931.0   \n",
       "3       TRBJ2-3*00            94|284|307|0|190||950.0   \n",
       "4       TRBJ2-6*00            94|286|307|0|192||960.0   \n",
       "...            ...                                ...   \n",
       "249928  TRBJ2-3*00  96|288|313|0|194|I102AI102C|936.0   \n",
       "249929  TRBJ1-4*00       94|279|307|0|185|ST97A|911.0   \n",
       "249930  TRBJ2-5*00            94|289|310|0|195||975.0   \n",
       "249931  TRBJ2-7*00            94|287|313|0|193||965.0   \n",
       "249932  TRBJ1-4*00      90|291|310|0|201|SA106G|991.0   \n",
       "\n",
       "                                       allDAlignments  \\\n",
       "0       12|17|36|197|202||25.0;16|21|48|197|202||25.0   \n",
       "1                              26|31|48|186|191||25.0   \n",
       "2                                                 NaN   \n",
       "3                              22|30|48|194|202||40.0   \n",
       "4                              29|34|48|192|197||25.0   \n",
       "...                                               ...   \n",
       "249928  19|24|36|194|199||25.0;29|34|48|195|200||25.0   \n",
       "249929                                            NaN   \n",
       "249930                                            NaN   \n",
       "249931                           1|9|36|193|201||40.0   \n",
       "249932                                            NaN   \n",
       "\n",
       "                           allJAlignments  \n",
       "0                 26|70|70|205|249||220.0  \n",
       "1                 23|71|71|193|241||240.0  \n",
       "2                 24|71|71|213|260||235.0  \n",
       "3       21|69|69|204|252|SA61CSC67G|212.0  \n",
       "4                 24|73|73|197|246||245.0  \n",
       "...                                   ...  \n",
       "249928       24|69|69|201|246|SC67G|211.0  \n",
       "249929            30|71|71|188|229||205.0  \n",
       "249930       24|68|68|205|249|SC66G|206.0  \n",
       "249931            26|59|67|211|244||165.0  \n",
       "249932            30|71|71|212|253||205.0  \n",
       "\n",
       "[249933 rows x 126 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
